src/dataflow/operators/ros/to_ros_operator.rs

```diff
@@ -0,0 +1,76 @@
+use crate::dataflow::{context::SinkContext, operator::Sink, Data};
+use serde::Deserialize;
+use std::sync::Arc;
+
+/// Takes an input ERDOS stream and publishes to a ROS topic using the provided message conversion
+/// function.
+///
+/// Conversion function should convert a Rust data type and return a ROS type which implements
+/// the [`rosrust::Message`] trait.
+///
+/// # Example
+/// The following example shows how to use a [`ToRosOperator`] with a conversion function which takes
+/// a Rust [`i32`] and converts it to a ROS message with [`rosrust_msg::std_msgs::Int32`] data.
+///
+/// Assume that `source_stream` is an ERDOS stream sending the correct messages.
+///
+/// ```
+/// fn erdos_int_to_ros_int(input: &i32) -> rosrust_msg::std_msgs::Int32 {
+///     rosrust_msg::std_msgs::Int32 { data: input.data }
+/// }
+///
+/// let ros_sink_config = OperatorConfig::new().name("ToRosInt32");
+/// erdos::connect_sink(
+///     move || -> ToRosOperator<i32, rosrust_msg::std_msgs::Int32> {
+///         ToRosOperator::new("int_topic", erdos_int_to_ros_int)
+///     },
+///     || {},
+///     ros_sink_config,
+///     &source_stream,
+/// );
+/// ```
+
+pub struct ToRosOperator<T, U: rosrust::Message>
+where
+    T: Data + for<'a> Deserialize<'a>,
+{
+    publisher: rosrust::Publisher<U>,
+    to_ros_msg: Arc<dyn Fn(&T) -> U + Send + Sync>,
```

Comment:
I think we should expose the timestamp to the `to_ros_msg` function. Many ROS types have a separate timestamp field, and we want to encourage the flow of timestamps to ROS too.

Developers can just disregard it if they don't need it. But right now, they're gonna have to duplicate the timestamp in the data to get access to it in the `to_ros_msg` function.

Reply:
Agree with this. Can we change the `to_ros_msg` type to take ERDOS messages, e.g. `Fn(&Message<T>) -> U`?